### PR TITLE
[Ruby 3.0] Switch back to the original commander gem and update dependencies

### PR DESCRIPTION
**Pull request by ainame**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

As per https://github.com/fastlane/fastlane/issues/17931, I've been doing the migration to get Ruby 3.0 supported in fastlane. I recognised that `highline` gem is a blocker for that as it has invalid usage of Ruby 3.0. We need to update that library version to 2.x (is 1.7.2 right now). To do so, we also need to update another dependency `commander-fastlane` that also depends on `highline`. Due to historical reasons, we switch the use of the original `commander` to the forked one.

It tweaks its styling in the help banner specifically for how to indicate "default command" amongst subcommand options. https://github.com/fastlane/commander/pull/1

For example, `(* default)` and `*` asterisk are the ones we have added to the forked one.

```
% bundle exec fastlane --help
[✔] 🚀
  fastlane

  CLI for 'fastlane' - The easiest way to automate beta deployments and releases for your iOS and Android apps

        Run using `fastlane [platform] [lane_name]`
        To pass values to the lanes use `fastlane [platform] [lane_name] key:value key2:value2`

  Commands: (* default)
    action                 Shows more information for a specific command
    actions                Lists all available fastlane actions
    add_plugin             Add a new plugin to your fastlane setup
    docs                   Generate a markdown based documentation based on the Fastfile
    enable_auto_complete   Enable tab auto completion
    env                    Print your fastlane environment, use this when you submit an issue on GitHub
    help                   Display global or [command] help documentation
    init                   Helps you with your initial fastlane setup
    install_plugins        Install all plugins for this project
    lanes                  Lists all available lanes and shows their description
    list                   Lists all available lanes without description
    new_action             Create a new custom action for fastlane.
    new_plugin             Create a new plugin that can be used with fastlane
    run                    Run a fastlane one-off action without a full lane
    search_plugins         Search for plugins, search query is optional
    socket_server          Starts local socket server and enables only a single local connection
    trigger              * Run a specific lane. Pass the lane name and optionally the platform first.
    update_fastlane        Update fastlane to the latest release
    update_plugins         Update all plugin dependencies
```

This is all the fork has. After digging through the gem, I found the official way to customise this help description injecting a custom template to be rendered to `commander`. That way we lose the reason to maintain the forked repo anymore. This PR ported the customisation to fastlane's codebase and stopped using the forked gem.

### Description

~This PR focuses on switching back to the original `commander` so, in the gemspec file, I locked the gem version to 4.4.4 which is equivalent to the head of the forked one. (When the version was 4.4.3, it was forked and 4.4.3 and was bumped to 4.4.6 with some warning fixes back ported https://github.com/fastlane/commander/pull/6)~

~Next PR will focus on `commander` and `highline` to be updated to get rid of Ruby's deprecation warnings.~
https://github.com/fastlane/fastlane/pull/18622

It turned out that there isn't really change needed to migrate `commander` and `highline` other than a minor fix for tests. So this PR also bumps the version to the latest version on both gems.

https://github.com/commander-rb/commander/blob/master/History.rdoc 
https://github.com/JEG2/highline/pull/248#issuecomment-829220759

### Testing Steps

See if `bundle exec fastlane --help` or `bundle exec fastlane MODULE --help` looks exactly the same as before.

### CODE REVIEW

**Comment by ainame** [comment]
I managed to replicate the same behaviour without forking but with only a few additions using the official way to customise `commander` gem🎉 
I will add tests later.

**Review by minuscorp — COMMENTED** [review]
Pretty dope!! Let's see if this fixed some bugs with the forked version. Great work, as always🚀

**Comment by ainame** [comment]
@minuscorp Thank you for your reviews.

> Let's see if this fixed some bugs with the forked version

As I mentioned, I suppose this doesn't change much in terms of its behaviour🙂 Will work on version bump change soon 🚀

File: fastlane/lib/fastlane/commands_generator/help.erb
Comment (ainame) [inline fastlane/lib/fastlane/commands_generator/help.erb:1]: This file was brought from here.
https://github.com/fastlane/commander/blob/master/lib/commander/help_formatters/terminal_compact/help.erb

I also did some tweaks to substitute these helpers with inline code.
https://github.com/fastlane/commander/blob/master/lib/commander/runner.rb#L200-L223

**Comment by minuscorp** [comment]
That's right, maybe a intermediate step would be to just bump the minor version which _shouldn't_ break compatibility to see if that solves the issue, I can try it later on when this gets merged 🚀

**Comment by ainame** [comment]
The thing is that commander bumped highline's version in a patch version change at 4.4.**7**😅  https://github.com/commander-rb/commander/blob/master/History.rdoc#447--2018-10-22-

It doesn't matter as Bundler won't try to update to 4.4.7 but feels like using the equivalent version as before would be safer for now.

**Comment by ainame** [comment]
I needed to apply `program :help_formatter, Fastlane::CommandsGenerator::HelpFormatter` to other commands😅 Will do that.

**Comment by minuscorp** [comment]
🤦‍♂️ didn't caught that

**Review by joshdholtz — CHANGES_REQUESTED** [review]
Super excited about this change! It looks like `fastlane deliver download_metadata --help` errors out though 🤔 

```rb
bundler: failed to load command: fastlane (/Users/josh/.gem/ruby/2.6.6/bin/fastlane)
/Users/josh/Projects/fastlane/fastlane/fastlane_core/lib/fastlane_core/ui/help_formatter.rb:9:in `read': No such file or directory @ rb_sysopen - /Users/josh/Projects/fastlane/fastlane/fastlane_core/lib/fastlane_core/ui/command_help.erb (Errno::ENOENT)
        from /Users/josh/Projects/fastlane/fastlane/fastlane_core/lib/fastlane_core/ui/help_formatter.rb:9:in `template'
        from /Users/josh/.gem/ruby/2.6.6/gems/commander-4.6.0/lib/commander/help_formatters/terminal.rb:13:in `render_command'
        from /Users/josh/.gem/ruby/2.6.6/gems/commander-4.6.0/lib/commander/runner.rb:304:in `block (2 levels) in create_default_commands'
        from /Users/josh/.gem/ruby/2.6.6/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
        from /Users/josh/.gem/ruby/2.6.6/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
        from /Users/josh/Projects/fastlane/fastlane/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:43:in `block in run!'
        from /Users/josh/.gem/ruby/2.6.6/gems/commander-4.6.0/lib/commander/runner.rb:398:in `block in global_option_proc'
        from /Users/josh/.rubies/ruby-2.6.6/lib/ruby/2.6.0/optparse.rb:1582:in `block in parse_in_order'
        from /Users/josh/.rubies/ruby-2.6.6/lib/ruby/2.6.0/optparse.rb:1568:in `catch'
        from /Users/josh/.rubies/ruby-2.6.6/lib/ruby/2.6.0/optparse.rb:1568:in `parse_in_order'
        from /Users/josh/.rubies/ruby-2.6.6/lib/ruby/2.6.0/optparse.rb:1562:in `order!'
        from /Users/josh/.rubies/ruby-2.6.6/lib/ruby/2.6.0/optparse.rb:1656:in `permute!'
        from /Users/josh/.rubies/ruby-2.6.6/lib/ruby/2.6.0/optparse.rb:1678:in `parse!'
        from /Users/josh/.gem/ruby/2.6.6/gems/commander-4.6.0/lib/commander/runner.rb:379:in `parse_global_options'
        from /Users/josh/Projects/fastlane/fastlane/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:50:in `run!'
        from /Users/josh/.gem/ruby/2.6.6/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
        from /Users/josh/Projects/fastlane/fastlane/deliver/lib/deliver/commands_generator.rb:187:in `run'
        from /Users/josh/Projects/fastlane/fastlane/deliver/lib/deliver/commands_generator.rb:18:in `start'
        from /Users/josh/Projects/fastlane/fastlane/fastlane/lib/fastlane/cli_tools_distributor.rb:114:in `take_off'
        from /Users/josh/Projects/fastlane/fastlane/bin/fastlane:23:in `<top (required)>'
        from /Users/josh/.gem/ruby/2.6.6/bin/fastlane:23:in `load'
        from /Users/josh/.gem/ruby/2.6.6/bin/fastlane:23:in `<top (required)>'
        from /Users/josh/.gem/ruby/2.6.6/gems/bundler-2.2.16/lib/bundler/cli/exec.rb:63:in `load'
        from /Users/josh/.gem/ruby/2.6.6/gems/bundler-2.2.16/lib/bundler/cli/exec.rb:63:in `kernel_load'
        from /Users/josh/.gem/ruby/2.6.6/gems/bundler-2.2.16/lib/bundler/cli/exec.rb:28:in `run'
        from /Users/josh/.gem/ruby/2.6.6/gems/bundler-2.2.16/lib/bundler/cli.rb:494:in `exec'
        from /Users/josh/.gem/ruby/2.6.6/gems/bundler-2.2.16/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
        from /Users/josh/.gem/ruby/2.6.6/gems/bundler-2.2.16/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
        from /Users/josh/.gem/ruby/2.6.6/gems/bundler-2.2.16/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
        from /Users/josh/.gem/ruby/2.6.6/gems/bundler-2.2.16/lib/bundler/cli.rb:30:in `dispatch'
        from /Users/josh/.gem/ruby/2.6.6/gems/bundler-2.2.16/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
        from /Users/josh/.gem/ruby/2.6.6/gems/bundler-2.2.16/lib/bundler/cli.rb:24:in `start'
        from /Users/josh/.gem/ruby/2.6.6/gems/bundler-2.2.16/exe/bundle:49:in `block in <top (required)>'
        from /Users/josh/.gem/ruby/2.6.6/gems/bundler-2.2.16/lib/bundler/friendly_errors.rb:130:in `with_friendly_errors'
        from /Users/josh/.gem/ruby/2.6.6/gems/bundler-2.2.16/exe/bundle:37:in `<top (required)>'
        from /Users/josh/.gem/ruby/2.6.6/bin/bundle:23:in `load'
        from /Users/josh/.gem/ruby/2.6.6/bin/bundle:23:in `<main>'
```

File: fastlane_core/lib/fastlane_core/ui/help_formatter.rb
Comment (minuscorp) [inline fastlane_core/lib/fastlane_core/ui/help_formatter.rb:9]: I'm wondering which directory is being loaded in the the failing case @joshdholtz

**Comment by minuscorp** [comment]
`fastlane/fastlane_core/lib/fastlane_core/ui/command_help.erb` this is not the right path isn't it? But why for that case, or you got the error in other subcommands?

**Comment by minuscorp** [comment]
The file is needed `help.erb` instead of `command-help.erb` as far as I can see, isn't it?

**Comment by ainame** [comment]
@joshdholtz Oh... Nice catch, thanks!

@minuscorp This is due to that there are two types of help. One is the global one (for the top-level command) the other is specific for a subcommand. 
https://github.com/commander-rb/commander/blob/master/lib/commander/help_formatters/terminal.rb#L12-L14


This seemed to fail with the latter one because of the subclass that working out the file path based on the current file's path. I didn't prepare for command_help.erb. I could either copy "command_help.erb" or delegate the rendering to the super class? Will have a look.

**Comment by ainame** [comment]
@joshdholtz @minuscorp Fixed the issue😄

File: .circleci/config.yml
Comment (ainame) [inline .circleci/config.yml:94]: Since we started using orbs for shellcheck, normal testing doesn't need homebrew.

File: .circleci/config.yml
Comment (ainame) [inline .circleci/config.yml:179]: However, fastlane.swift uses swiftformat installed via `brew bundle`. This would need this debug info.

File: .circleci/config.yml
Comment (ainame) [inline .circleci/config.yml:182]: I believe we normally don't need this `brew update`...
https://github.com/fastlane/fastlane/pull/17863

I don't know why it failed here though. If the brew version matters, using a newer macOS image might (or might not) help it 🤔 
https://app.circleci.com/pipelines/github/fastlane/fastlane/2629/workflows/c48b73e5-02a7-4563-bf8f-8e19cbb6e508/jobs/66023

File: .circleci/config.yml
Comment (ainame) [inline .circleci/config.yml:268]: Because of this reason, I updated used Xcode and Ruby versions here.
https://github.com/fastlane/fastlane/pull/18599/files#r624535522

Comment (minuscorp) [reply .circleci/config.yml:268]: At least we're using Xcode 12.4 for latest Swift releases if you want another version bump

Comment (ainame) [reply .circleci/config.yml]: Bumped to 12.4 😄

**Review by joshdholtz — APPROVED** [review]
The subcommands now work! This is so 🔥 This looks good to me ☺️ I'll let  @minuscorp tdo another view on this before I click the merge button! Thanks for getting us back on track here ❤️

**Review by minuscorp — APPROVED** [review]
This is great to me. Great addition and less deprecated versions to maintain 🚀

**Comment by ainame** [comment]
Let's archive this 🙂  https://github.com/fastlane/commander

**Comment by rogerluan** [comment]
> Let's archive this 🙂 https://github.com/fastlane/commander

💯 ☝️ 

@joshdholtz I think you're the only one with permission to do so? 😊

**Pull request merged**
